packages/loader/driver-definitions/src/urlResolver.ts

```diff
@@ -38,6 +38,26 @@ export interface IUrlResolver {
     ): Promise<string>;
 }
 
+/**
+* Information that can be returned by a lightweight, seperately exported driver function. Used to preanalyze a URL
+* for driver compatibility and preload information.
+*/
+export interface DriverPreCheckInfo {
+    /**
+     * Only needed as long as long as Loader.request() does not work as intended. When Loader.request() caches and
+     * resolves pathing properly, this can be removed. #4489, #4491
+     */
+    componentPath: string;
+    /**
+     * Container name, if known. This could be used to eagerly prefetch container code prior to having a snapshot.
+     */
+    containerCodeHint?: string;
+    /**
+     * Domains that will be connected to on the critical boot path. Can be preconnected to in certain flows.
+     */
+    preconnectDomains?: string[];
```

Comment:
can you expand on this more. i'm not really sure what the content here is, or how it's used

Reply:
It's a list of domains, like sharepoint.com that a driver knows it will connect to on the critical path. Then a host can add preconnect information to the page's HTML contents to "preconnect" to that domain, improving load times. For example, you can see how we add it to the IFrame HTML in srcdocGenerator.ts. Christian can probably give even more details. :)

Reply:
What do you think of renaming this to criticalBootDomains with one possible use case being preconnect but leaving the naming open for other similar uses cases too? Not sure what those would be... but I'm sure the internet will come up with something.

Reply:
criticalBootDomains sounds better to me as it may be used to serve other purposes too in future.

Reply:
you got it :D